lms/djangoapps/courseware/user_state_client.py

```diff
@@ -166,16 +209,40 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
                 },
             )
 
+            num_fields_before = num_fields_after = num_new_fields_set = len(state)
+            num_fields_updated = 0
             if not created:
                 if student_module.state is None:
                     current_state = {}
                 else:
                     current_state = json.loads(student_module.state)
+                num_fields_before = len(current_state)
                 current_state.update(state)
+                num_fields_after = len(current_state)
```

Comment:
I'll add comments to the lines below. If someone deletes that code, the remaining variables will be pep/pylint violations.